test/Libraries/Microsoft.Extensions.TimeProvider.Testing.Tests/FakeTimeProviderTimerTests.cs

```diff
@@ -319,13 +324,13 @@ public void MultipleTimersCallbackInvokedInScheduledOrder()
 
         Assert.Equal(new[]
         {
-            (1, TimeSpan.FromMilliseconds(3)),
-            (2, TimeSpan.FromMilliseconds(3)),
-            (3, TimeSpan.FromMilliseconds(6)),
-            (1, TimeSpan.FromMilliseconds(6)),
-            (2, TimeSpan.FromMilliseconds(6)),
-            (1, TimeSpan.FromMilliseconds(9)),
-            (2, TimeSpan.FromMilliseconds(9)),
+            (1, TimeSpan.FromMilliseconds(11)),
+            (2, TimeSpan.FromMilliseconds(11)),
+            (3, TimeSpan.FromMilliseconds(11)),
+            (1, TimeSpan.FromMilliseconds(11)),
+            (2, TimeSpan.FromMilliseconds(11)),
+            (1, TimeSpan.FromMilliseconds(11)),
+            (2, TimeSpan.FromMilliseconds(11)),
             (3, TimeSpan.FromMilliseconds(11)),
```

Comment:
Why is the assertion changed here? The point of the test is to verify that invocations happen in order and at the expected time.

Reply:
The clock advanced by 11 milliseconds, why should the callbacks be annotated with intermediate time stamps? If the timer advanced in smaller increments, then the behavior would be reasonable.

Reply:
This particular scenario is described in https://github.com/dotnet/extensions/issues/3995 and was fixed in https://github.com/dotnet/extensions/pull/4022.

In essence, this test verifies that the result of `GetUtcNow` was as expected each time the timer's callbacks were invoked, as well as the order the timers were invoked in.

Granted, this test should be split up in two, one for each condition, to make it clear to all what is going on, that's on me.

Reply:
The test has an incorrect assumption that the time at which a timer callback is invoked corresponds to its requested wake time. This was an implementation artifact of the previous design and is not a contractual guarantee of the ITimer interface. The guarantee is that the callbacks will not be invoked before the due time, but invariable are invoked later than the due time.

I've updated the test to increment the clock gradually, which makes it so it satisfies the previously enforced conditions since the callbacks are invoked gradually. But this is still assuming that the timer behaves in a way that is not actually guaranteed by the API contract. I'm OK leaving the test this way however, it's not a big deal.

Reply:
> The test has an incorrect assumption that the time at which a timer callback is invoked corresponds to its requested wake time. This was an implementation artifact of the previous design and is not a contractual guarantee of the ITimer interface. The guarantee is that the callbacks will not be invoked before the due time, but invariable are invoked later than the due time.

Sure, I know the production timer only gives us this guarantee, however, the FakeTimeProvider is deterministic (should be) and I see no reason it should not behave like this. 

This is the strength of the TimeProvider abstraction, that I can move time forward and have both a timer callback and a "UTC now" be in sync, yielding predictable tests. If the timer callback and the current time are not in sync, we may as well have an `IClock` interface that only exposes `GetUtcNow` and a `ITimer` interface that does its thing.